atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java

```diff
@@ -337,8 +336,16 @@ void setServersInPoolTo(ImmutableSet<CassandraServer> desiredServers) {
         Set<CassandraServer> serversToShutdown = absentHostTracker.incrementAbsenceAndRemove();
 
         if (!(serversToAdd.isEmpty() && absentServers.isEmpty())) { // if we made any changes
-            sanityCheckRingConsistency();
-            cassandra.refreshTokenRangesAndGetServers();
+            Set<CassandraServer> invalidServers =
```

Comment:
So, this is _more_ than a bit sketch, and I'd really want to refactor this so that we've a `staging` for cluster containers or something. That said, I'll leave it to the reviewer to decide if this is reasonable. But as it stands, there's a race condition here in that, between validating the pools, a host that has not been validated could be picked, if for whatever reason we all of our hosts have been blocklisted. _however_, this is the present case currently, but we should aim to be better.